lib/puppet/type/file/content_uri.rb

```diff
@@ -0,0 +1,42 @@
+require 'net/http'
+require 'uri'
+
+module Puppet
+  Puppet::Type.type(:file).newparam(:content_uri) do
+
+    desc <<-'EOT'
+      A unique identifier for file content, which must be a URI which points
+      to a file in a module. This parameter allows the content of a file resource
+      to remain consistent regardless of code updates on the master.
```

Comment:
> Also source file updates for things in the puppet file server?

Yeah not sure how to describe it. The term `code updates on the master` is a little confusing because it almost sounds like the server's clojure or ruby code. It's more the content of files managed by the puppet fileserver, and our consistency guarantees only apply when the file content is within a per-environment directory (eg `$codedir/environments/<env>`) backed by version control (built-in PE or wired up in FOSS).